src/java.base/share/classes/jdk/internal/util/ReferencedKeySet.java

```diff
@@ -75,6 +75,20 @@ public final class ReferencedKeySet<T> extends AbstractSet<T> {
      */
     final ReferencedKeyMap<T, ReferenceKey<T>> map;
 
+    /**
+     * @return a supplier to create a {@code ConcurrentHashMap} appropriate for use in the
+     *         create methods.
+     * @param <E> the type of elements maintained by this set
+     */
+    public static <E> Supplier<Map<ReferenceKey<E>, ReferenceKey<E>>> concurrentHashMapSupplier() {
+        return new Supplier<>() {
```

Comment:
Can this just `return ReferencedKeyMap.concurrentHashMapSupplier();`?

Reply:
It compiles, so maybe. :-) Running some tests.